### PR TITLE
Add simple CRUD for bucket

### PR DESCRIPTION
**Pull request by nikos83**

### CODE REVIEW

File: app/controllers/buckets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        if @bucket.update(bucket_params)
+          redirect_to root_path, notice: 'Updated!'
+        else
+          render :edit
+        end
+      end
+    
+      def destroy
+        if @bucket.destroy
+          redirect_to root_path, notice: 'Destoyed!'
```
Comment (zuzannast) [inline app/controllers/buckets_controller.rb]: typo "Destroyed!"

File: app/controllers/buckets_controller.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+class BucketsController < ApplicationController
+  before_action :set_bucket, only: [:show, :edit, :update, :destroy]
+    def index
+        @buckets = Bucket.all
```
Comment (zuzannast) [inline app/controllers/buckets_controller.rb]: Could you use two spaces for indents? 😄

File: app/views/buckets/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<h1>Editing Bucket</h1>
+
+<%= render 'form', bucket: @bucket %>
+
+<%= link_to 'Show', @bucket %> |
```
Comment (zuzannast) [inline app/views/buckets/edit.html.erb]: I think this should be `link_to "Show", bucket_path(@bucket)`

File: app/views/buckets/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      <th colspan="3"></th>
+    </tr>
+  </thead>
+ 
+  <tbody>
+    <% @buckets.each do |bucket| %>
+      <tr>
+        <td><%= bucket.name %></td>
+        <td><%= bucket.user_id %></td>
+        <td><%= link_to "Show", bucket %></td>
```
Comment (zuzannast) [inline app/views/buckets/index.html.erb]: I think this should be `link_to "Show", bucket_path(@bucket)`

File: app/controllers/buckets_controller.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
+class BucketsController < ApplicationController
+  before_action :set_bucket, only: [:show, :edit, :update, :destroy]
+    def index
```
Comment (tomajask) [inline app/controllers/buckets_controller.rb]: Empty line would be nice here :)

File: app/controllers/buckets_controller.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    end
+
+    def create
+      @bucket = current_user.buckets.new(bucket_params)
+      if @bucket.save
+        redirect_to root_path, notice: 'CREATED!'
+      else
+        render :new
+      end
+    end
```
Comment (tomajask) [inline app/controllers/buckets_controller.rb]: Empty line would be nice here :)

File: app/controllers/buckets_controller.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    def update
+      if @bucket.update(bucket_params)
+        redirect_to root_path, notice: 'Updated!'
+      else
+        render :edit
+      end
+      end
```
Comment (tomajask) [inline app/controllers/buckets_controller.rb]: Indentation inconsistent :)

File: app/controllers/buckets_controller.rb [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      private
+      def set_bucket
+        @bucket = Bucket.find(params[:id])
+      end
+    
+      def bucket_params
+        params.require(:bucket).permit(:name, :user_id)
+      end
+    
+
```
Comment (tomajask) [inline app/controllers/buckets_controller.rb]: too many empty lines :)

**Pull request merged**